hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/balancer/FindTargetGreedy.java

```diff
@@ -60,17 +61,27 @@ public FindTargetGreedy(
     this.containerManager = containerManager;
     this.placementPolicy = placementPolicy;
     this.nodeManager = nodeManager;
+
+    potentialTargets = new TreeSet<>((a, b) -> {
+      double currentUsageOfA = a.calculateUtilization(
+          sizeEnteringNode.get(a.getDatanodeDetails()));
+      double currentUsageOfB = b.calculateUtilization(
+          sizeEnteringNode.get(b.getDatanodeDetails()));
+      return Double.compare(currentUsageOfA, currentUsageOfB);
```

Comment:
Sorry for not bringing this up earlier! But we will need to handle the case in the comparator where utilisation is same for two nodes.

Reply:
+1 o.w.

Reply:
good point , thanks for pointing out this !

Reply:
good point , thanks for pointing out this